api/new-contribution.ts

```diff
@@ -0,0 +1,143 @@
+      chunks.push(chunk.trim());
+      chunk = '';
+    }
+    chunk += ' ' + word;
+  }
+
+  if (chunk.length > 0) {
+    chunks.push(chunk.trim());
+  }
+
+  return chunks;
+}
+
+async function getSpamScore(title: string, body: string, threshold: number, tokenLimit: number) {
+  const paragraphs = body.split('\n\n'); // Splitting by paragraphs
+  for (const paragraph of paragraphs) {
+    const filteredParagraph = filterText(paragraph);
+    const subParagraphs = chunkText(filteredParagraph, tokenLimit - title.length);
+
+    for (const subParagraph of subParagraphs) {
+      try {
+        if (!SUBMIT_TO_OPENAI) {
+          console.log({role: 'user', content: `title: ${title}\npart of body: ${subParagraph}`});
+        }
+        
+        if (SUBMIT_TO_OPENAI) {
+          const chatCompletion = await openai.chat.completions.create({
+            messages: [
+              {
+                role: 'system',
+                content: `You are a helpful project gallery forum moderator. I will give you a title, and part of the body of a post, and you will return a rating of 1-7 of how likely it is to be spam, along with the reasons for why you either consider it safe or a high chance of it being spam as structured JSON format like {"rating": <number>, "reasons": ["reason1", "reason2"]}`,
+              },
+              {role: 'user', content: `title: ${title}\npart of body: ${subParagraph}`},
+            ],
+            model: 'gpt-3.5-turbo',
+          });
+
+          if (chatCompletion.choices?.[0]?.finish_reason === 'stop') {
+            const spamAnalysis = JSON.parse(chatCompletion?.choices[0]?.message?.content) || {
+              rating: 0,
+              reasons: [],
+            };
+
+            const {rating, reasons} = spamAnalysis;
+
+            if (rating >= threshold) {
+              return {stopEarly: true, rating, reasons};
+            }
+          }
+        }
+      } catch (e) {
+        console.log(e);
+
+        throw e
+      }
+    }
+  }
+  return {stopEarly: false, rating: 0, reasons: []}; // Use 0 or any other default value
+}
+
+export default async function (req: VercelRequest, res: VercelResponse) {
+  const document = req.body;
+
+  const title = document?.title;
+  const bodyMarkdown = toMarkdown(document.body, {});
+
+  const {rating, reasons} = await getSpamScore(title, bodyMarkdown, 4, TOKEN_LIMIT);
+
+  const curatedContribution = {
+    _type: 'curatedContribution',
+    approved: rating < 4,
+    contribution: {
+      _ref: document._id,
+      _type: 'reference',
+      _weak: true,
+    },
+    spamRating: rating,
+    approvalReasons: reasons,
+  };
+
+  if (WRITE_TO_SANITY) {
+    await writeClient.createIfNotExists(curatedContribution);
+  }
+
+  if (POST_TO_SLACK) {
+    const slackMessage = {
+      text: `New contribution received: ${title}.\nSpam Rating: ${rating}\nReasons: ${reasons.join(', ')}`,
+    };
+
+    await axios.post(slackWebhookUrl, slackMessage);
+  }
```

Comment:
We (that is, I) can easily replace this with a GROQ-powered Webhook if we want to reduce complexity.